### PR TITLE
Switch CI to pytorch 1.13

### PR DESCRIPTION
**Pull request by danthe3rd**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #521

Closes https://github.com/facebookresearch/xformers/issues/515
Closes https://github.com/facebookresearch/xformers/issues/514

Note:
`static_argnums`'s argument to `memory_efficient_fusion` is now removed, so had to update some code

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -542,21 +544,25 @@ workflows:
...
+          pytorch_version: "1.12.1"
       - build_conda:
-          name: conda_build_cu116
+          name: conda_build_cu116_1.12.1
           dockerimage: pytorch/conda-builder:cuda116
           cu_version: "11.6"
-          filters:
-            branches:
-              only:
-                - main
```
Comment (danthe3rd) [inline .circleci/config.yml:559]: XXX: Need to revert this before merging (only run on main branch)

File: .circleci/config.yml [hunk 2]
```diff
@@ -396,8 +398,8 @@ jobs:
           no_output_timeout: 20m
           command: |
             git submodule update --init --recursive
-            python packaging/conda/build_conda.py  --cuda << parameters.cu_version >> --python 3.9 --pytorch 1.12.1 --upload-dev
-            python packaging/conda/build_conda.py  --cuda << parameters.cu_version >> --python 3.10 --pytorch 1.12.1 --upload-dev
+            python packaging/conda/build_conda.py --cuda << parameters.cu_version >> --python 3.9 --pytorch << parameters.pytorch_version >>
+            python packaging/conda/build_conda.py --cuda << parameters.cu_version >> --python 3.10 --pytorch << parameters.pytorch_version >>
```
Comment (danthe3rd) [inline .circleci/config.yml:402]: XXX: Need to put back "--upload-dev" before merging

File: requirements-test.txt [hunk 3]
```diff
@@ -27,4 +27,5 @@ hydra-core >= 1.1
 fairscale >= 0.4.5
 
 # Dependency for fused layers, optional
-triton == 2.0.0.dev20220701
+triton == 2.0.0.dev20221105
+networkx
```
Comment (fmassa) [inline requirements-test.txt:31]: nit: is this necessary?

File: requirements-test.txt [hunk 4]
```diff
@@ -27,4 +27,5 @@ hydra-core >= 1.1
 fairscale >= 0.4.5
 
 # Dependency for fused layers, optional
-triton == 2.0.0.dev20220701
+triton == 2.0.0.dev20221105
```
Comment (fmassa) [inline requirements-test.txt:30]: We will have to get @blefaudeux PR with Triton changes in before merging this

**Review by fmassa — APPROVED** [review]
LGTM, thanks!

We will need to get Triton version updated first before merging this, but maybe there is a way of getting this merged before that?

File: requirements-test.txt (hunk 4 above)
Comment (blefaudeux) [reply requirements-test.txt:30]: should be finally really to merge, sorry for the delay I've been a little underwater with some projects..

**Pull request merged**